cli/command/context/create.go

```diff
@@ -62,8 +62,11 @@ func newCreateCommand(dockerCli command.Cli) *cobra.Command {
 		&opts.DefaultStackOrchestrator,
 		"default-stack-orchestrator", "",
 		"Default orchestrator for stack operations to use with this context (swarm|kubernetes|all)")
+	flags.SetAnnotation("default-stack-orchestrator", "deprecated", nil)
 	flags.StringToStringVar(&opts.Docker, "docker", nil, "set the docker endpoint")
 	flags.StringToStringVar(&opts.Kubernetes, "kubernetes", nil, "set the kubernetes endpoint")
+	flags.SetAnnotation("kubernetes", "kubernetes", nil)
+	flags.SetAnnotation("kubernetes", "deprecated", nil)
```

Comment:
Curious; did you also try if `flags.MarkDeprecated("kubernetes", "<deprecation message>")` would do the trick? 

Given: that also hides the flag, and prints a deprecation message when used, so it would have a slightly different result; https://github.com/spf13/pflag/blob/85dd5c8bc61cfa382fecd072378089d4e856579d/flag.go#L408-L422

```go
// MarkDeprecated indicated that a flag is deprecated in your program. It will
// continue to function but will not show up in help or usage messages. Using
// this flag will also print the given usageMessage.
func (f *FlagSet) MarkDeprecated(name string, usageMessage string) error {
	flag := f.Lookup(name)
	if flag == nil {
		return fmt.Errorf("flag %q does not exist", name)
	}
	if usageMessage == "" {
		return fmt.Errorf("deprecated message for flag %q must be set", name)
	}
	flag.Deprecated = usageMessage
	flag.Hidden = true
	return nil
}
```

Reply:
> but will not show up in help or usage messages

Do we really want that?

I was expecting `flags.SetAnnotation("kubernetes", "deprecated", nil)` to actually end up marking the flag as deprecated in the docs, a bit like the `"kubernetes"` annotation (sorry it doesn’t help that here it’s both a flag and an annotation 😬) in https://docs.docker.com/engine/reference/commandline/stack/#options
I think it has a good chance to work according to the code for the `deprecated-badge` and `kubernetes` at https://github.com/docker/docker.github.io/blob/master/_includes/cli.md#options

Reply:
> Do we really want that?

Gotcha.

Normally, yes (deprecate in "major" release X, and remove in "major release X+n"),  but given that we're "cheating" a bit, and do this in a patch release, I it may indeed be a bit unexpected.